datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -3206,6 +3261,16 @@ mod tests {
         make_array(&args).expect("failed to initialize function array")
     }
 
+    fn return_large_array() -> ArrayRef {
+        // Returns: [1, 2, 3, 4]
+        let capacity = i32::MAX as usize + 10;
+        let args = vec![Arc::new(Int64Array::from(vec![Some(1)])) as ArrayRef; capacity];
+
+        println!("args.len() = {}", args.len());
+
+        make_array(&args).expect("failed to initialize function array")
+    }
+
```

Comment:
I don't know how to efficiently create a `LargeList` using `make_array` as it runs out of memory on my end.

Reply:
maybe you should use below constructor to construct LargeList
```rust
let list_array = Arc::new(LargeListArray::from_iter_primitive::<Int32Type, _, _>(
        vec![
            Some(vec![Some(0), Some(1), Some(2)]),
            None,
            Some(vec![Some(3), None, Some(5)]),
        ],
    )) as ArrayRef

```